ultravox/tools/tts.py

```diff
@@ -38,10 +88,64 @@ def tts(self, text: str, voice: Optional[str] = None, sample_rate: int = 16000):
             "User-Agent": "MyTTS",
         }
         body = _make_ssml(voice, text)
-        response = self._session.post(url, headers=headers, data=body)
-        response.raise_for_status()
+        return self._handle_pcm_response(self._post(url, headers, body))
 
-        pcm_array = np.frombuffer(response.content, dtype=np.int16)
-        wav_bytes = io.BytesIO()
-        sf.write(wav_bytes, pcm_array, sample_rate, format="wav")
-        return wav_bytes.getvalue()
+
+class ElevenTts(Client):
+    DEFAULT_VOICE = "21m00Tcm4TlvDq8ikWAM"
+    DEFAULT_MODEL = "eleven_multilingual_v2"
+    ALL_VOICES = [
+        "21m00Tcm4TlvDq8ikWAM",
+        "29vD33N1CtxCmqQRPOHJ",
+        "2EiwWnXFnvU5JabPnv8n",
+        "5Q0t7uMcjvnagumLfvZi",
+        "AZnzlk1XvdvUeBnXmlld",
+        "CYw3kZ02Hs0563khs1Fj",
+        "D38z5RcWu1voky8WS1ja",
+        "EXAVITQu4vr4xnSDxMaL",
+        "ErXwobaYiN019PkySvjV",
+        "GBv7mTt0atIp3Br8iCZE",
+        "IKne3meq5aSn9XLyUdCD",
+        "JBFqnCBsd6RMkjVDRZzb",
+        "LcfcDJNUP1GQjkzn1xUU",
+        "MF3mGyEYCl7XYWbV9V6O",
+        "N2lVS1w4EtoT3dr4eOWO",
+        "ODq5zmih8GrVes37Dizd",
+        "SOYHLrjzK2X1ezoPC6cr",
+        "TX3LPaxmHKxFdv7VOQHJ",
+        "ThT5KcBeYPX3keUQqHPh",
+        "TxGEqnHWrfWFTfGW9XjX",
+        "VR6AewLTigWG4xSOukaG",
+        "XB0fDUnXU5powFXDhCwa",
+        "Xb7hH8MSUJpSbSDYk0k2",
+        "XrExE9yKIg1WjnnlVkGX",
+        "ZQe5CZNOzWyzPSCn5a3c",
+        "Zlb1dXrM653N07WRdFW3",
+    ]
+
+    def tts(self, text: str, voice: Optional[str] = None):
+        voice = voice or self.DEFAULT_VOICE
+        if voice == RANDOM_VOICE_KEY:
+            # Every process has same random seed, so we mix in the PID here for more variation.
+            i = np.random.randint(len(self.ALL_VOICES)) + os.getpid()
+            voice = self.ALL_VOICES[i % len(self.ALL_VOICES)]
+        url = f"https://api.elevenlabs.io/v1/text-to-speech/{voice}/stream?output_format=pcm_16000"
+        print("url", url)
+        headers = {"xi-api-key": os.environ["ELEVEN_API_KEY"]}
+        body = {
+            "text": text,
+            "model_id": self.DEFAULT_MODEL,
+            "voice_settings": {
+                "stability": 0.5,
+                "similarity_boost": False,
+            },
+        }
+        return self._handle_pcm_response(self._post(url, headers, body))
+
+
+def create_client(implementation: str, sample_rate: int):
+    if implementation == "azure":
+        return AzureTts(sample_rate=sample_rate)
+    elif implementation == "eleven":
+        return ElevenTts(sample_rate=sample_rate)
```

Comment:
Nit: I'm wondering if we ever want to randomize between the two implementations. Not important though.